Userland/Libraries/LibWeb/CSS/StyleValue.cpp

```diff
@@ -299,15 +299,8 @@ BackgroundStyleValue::BackgroundStyleValue(
     NonnullRefPtr<StyleValue> attachment,
     NonnullRefPtr<StyleValue> origin,
     NonnullRefPtr<StyleValue> clip)
-    : StyleValue(Type::Background)
-    , m_color(color)
-    , m_image(image)
-    , m_position(position)
-    , m_size(size)
-    , m_repeat(repeat)
-    , m_attachment(attachment)
-    , m_origin(origin)
-    , m_clip(clip)
+    : StyleValueWithDefaultOperators(Type::Background)
+    , m_properties { color, image, position, size, repeat, attachment, origin, clip, 0 }
```

Comment:
I'm wondering if this would be better as
```c++
m_properties {
    .color = color,
    .image = image,
    ...
}
```
(And probably with `move()` as well, but that's another issue.)

It's verbose, but one issue I keep being uncomfortable about is that StyleValue arguments are generally all StyleValues, and so it'd be easy to put them in the wrong order. This wouldn't solve the issue of passing them to the constructor in the wrong order (~Jakt rewrite wen?~) but it would make sure that changing the Properties struct doesn't create bugs here.

Thoughts?

Reply:
Yeah I did think about that, but I decided not to because of the constructor thing (since it does not protect against that like you mention). But maybe it's worth doing despite the extra verbosity to remove one possible error source :smile: